### ISSUE TITLE
`IndexError` when `pull_state` is larger than 77 for `Genshin5starEPWeaponModel`

### ISSUE THREAD

**Issue by seriaati** [OneBST/GGanalysis#7]
The `Genshin5starEPWeaponModel` model has issues.  

Code:
```py
for i in range(81):
    try:
        # 武器池池在墊了 i 抽，有大保底，命定值為 0 的情況下抽 1 個UP五星抽數的預期抽數
        dist_w = GI.up_5star_ep_weapon(
            item_num=1, pull_state=i, up_guarantee=1, fate_point=0
        )
        print(i, dist_w.exp)
    except:
        pass
```

Result:
```py
0 96.5163329360168
1 95.88466398747471
2 95.24854218632458
3 94.6079361428703
4 93.96281424613886
5 93.31314466232071
...
76 44.265942350628215
77 nan
```

Issues noticed:
When `i=77`, a `RuntimeWarning` is raised, and `dist_w.expt` returns `nan`
```js
GGanalysis\distribution_1d.py:124: RuntimeWarning: invalid value encountered in divide
  return FiniteDist(ans/sum(ans))
```

When `i >= 78`, no output is printed because an `IndexError` is raised.
Traceback:
```js
File "C:\Users\seria\AppData\Local\Programs\Python\Python310\lib\site-packages\GGanalysis\games\genshin_impact\gacha_model.py", line 93, in __call__
    return super().__call__(item_num, multi_dist, pull_state, up_guarantee, fate_point, *args, **kwds)
  File "C:\Users\seria\AppData\Local\Programs\Python\Python310\lib\site-packages\GGanalysis\basic_models.py", line 30, in __call__
    return self._get_dist(item_num, parameter_list)
  File "C:\Users\seria\AppData\Local\Programs\Python\Python310\lib\site-packages\GGanalysis\basic_models.py", line 49, in _get_dist
    ans_dist = self._forward(parameter_list)
  File "C:\Users\seria\AppData\Local\Programs\Python\Python310\lib\site-packages\GGanalysis\basic_models.py", line 65, in _forward
    ans_dist = layer(ans_dist, *parameter[0], **parameter[1])
  File "C:\Users\seria\AppData\Local\Programs\Python\Python310\lib\site-packages\GGanalysis\gacha_layers.py", line 18, in __call__
    return self._forward(input, 1), self._forward(input, 0, *args, **kwds)
  File "C:\Users\seria\AppData\Local\Programs\Python\Python310\lib\site-packages\GGanalysis\gacha_layers.py", line 43, in _forward
    return cut_dist(self.dist, pull_state)
  File "C:\Users\seria\AppData\Local\Programs\Python\Python310\lib\site-packages\GGanalysis\distribution_1d.py", line 123, in cut_dist
    ans[0] = 0
IndexError: index 0 is out of bounds for axis 0 with size 0
```

If I remember correclty, the maximum `pull_state` (墊抽) one can have on weapon banners is 79, since on the 80th pull you'd get the 5 star.

**Comment by seriaati** [comment]
Oh, I have just realized.
I am guessing that the model is saying that 77 is the maximum pulls needed to get a 5 star banner weapon under any situation?

**Comment by OneBST** [comment]
Yes, the actual pity pos of genshin weapon banner is 77, instead of the claimed 80.
This conclusion is derived from statistical data. You can watch my video explain how I build this model on [bilibili](https://www.bilibili.com/video/BV19Y411W7W5/) or on [Youtube](https://www.youtube.com/watch?v=8ZxzVE_Pmtc).

**Comment by OneBST** [comment]
In addition, the actual pity pull of 4 star item in weapon pool is 9 which is also different from the officially claimed 10.

**Comment by seriaati** [comment]
What does `.exp` mean? I was assuming that it is the number of pulls needed to pull a 5 star banner weapon but seems like it's not?

**Comment by OneBST** [comment]
The `FiniteDisrt.exp` means the expectation (or average) pulls of getting this specific item.
In this case, the expectation pulls of getting a specific UP 5 star weapon is 105.67.
``` python
import GGanalysis.games.genshin_impact as GI
# This is expectation pulls without epitomized path (定轨) 133.12597646347143
print(GI.up_5star_specific_weapon(1).exp)
# This is expectation pulls with epitomized path (定轨) 105.66874381788045
print(GI.up_5star_ep_weapon(1).exp)
```
Expectation is not enough to describe the whole distribution. I recommand use graphs in [this articale of mine](https://www.bilibili.com/read/cv12616453).

**Comment by OneBST** [comment]
I add a documentation on this repositorie's github pages.
You can check [it](https://onebst.github.io/GGanalysis/index.html) out. It may helps.

**Comment by seriaati** [comment]
抱歉最近比較忙沒看 GitHub
感謝添加 documentation  幫了超級大忙
